src/state/apis/swappers/swappersApi.ts

```diff
@@ -1,9 +1,88 @@
 import { createApi } from '@reduxjs/toolkit/dist/query/react'
+import type { Result } from '@sniptt/monads'
+import { orderBy } from 'lodash'
+import type { GetTradeQuoteInput, SwapErrorRight, TradeQuote2 } from 'lib/swapper/api'
+import { SwapperName } from 'lib/swapper/api'
+import { getInputOutputRatioFromQuote } from 'state/apis/swappers/helpers/getInputOutputRatioFromQuote'
+import { getLifiTradeQuoteHelper } from 'state/apis/swappers/helpers/getLifiTradeQuoteApiHelper'
+import { getThorTradeQuoteHelper } from 'state/apis/swappers/helpers/getThorTradeQuoteApiHelper'
+import type { ApiQuote } from 'state/apis/swappers/types'
+import { isCrossAccountTradeSupported } from 'state/helpers'
+import type { ReduxState } from 'state/reducer'
+import type { FeatureFlags } from 'state/slices/preferencesSlice/preferencesSlice'
+import { selectFeatureFlags } from 'state/slices/selectors'
+import { tradeQuoteSlice } from 'state/slices/tradeQuoteSlice/tradeQuoteSlice'
 
 import { BASE_RTK_CREATE_API_CONFIG } from '../const'
 
 export const swappersApi = createApi({
   ...BASE_RTK_CREATE_API_CONFIG,
   reducerPath: 'swappersApi',
-  endpoints: _build => ({}),
+  endpoints: build => ({
+    getTradeQuote: build.query<ApiQuote[], GetTradeQuoteInput>({
+      queryFn: async (getTradeQuoteInput: GetTradeQuoteInput, { getState, dispatch }) => {
+        const queryStartTime = Date.now()
+        const state = getState() as ReduxState
+        const { sendAddress, receiveAddress } = getTradeQuoteInput
+        const isCrossAccountTrade = sendAddress !== receiveAddress
+        const { LifiSwap, ThorSwap }: FeatureFlags = selectFeatureFlags(state)
+        const quotes: (Result<TradeQuote2, SwapErrorRight> & {
+          swapperName: SwapperName
+        })[] = []
+        const quoteHelperArgs = [getTradeQuoteInput, state] as const
+        if (LifiSwap)
```

Comment:
This is where we'll add _n_ swappers, though we'll probably want a helper to make it tidier.